File: app/src/main/java/forpdateam/ru/forpda/ui/fragments/settings/BaseSettingFragment.java
package forpdateam.ru.forpda.ui.fragments.settings;

import android.os.Bundle;
import android.support.v7.preference.PreferenceFragmentCompat;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created by radiationx on 24.09.17.
 */

public class BaseSettingFragment extends PreferenceFragmentCompat {
    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {

    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        View rootView = getView();
        if (rootView != null) {
            RecyclerView list = (RecyclerView) rootView.findViewById(android.support.v7.preference.R.id.list);
            if (list != null) {
                list.setPadding(0, 0, 0, 0);
            }
        }
        setDividerHeight(0);
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/fragments/settings/BaseSettingFragment.kt
package forpdateam.ru.forpda.ui.fragments.settings

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceFragmentCompat
import androidx.recyclerview.widget.RecyclerView
import forpdateam.ru.forpda.R
import forpdateam.ru.forpda.extensions.getDimenPx

/**
 * Created by radiationx on 24.09.17.
 */

open class BaseSettingFragment : PreferenceFragmentCompat() {

    private var listScrollY = 0
    private var lastIsVisible = false

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {

    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view?.findViewById<RecyclerView>(androidx.preference.R.id.recycler_view)
            ?.also { list ->
                list.setPadding(0, 0, 0, 0)
                list.addOnScrollListener(object : RecyclerView.OnScrollListener() {
                    override fun onScrolled(
                        recyclerView: RecyclerView,
                        dx: Int,
                        dy: Int
                    ) {
                        super.onScrolled(recyclerView, dx, dy)
                        listScrollY = recyclerView.computeVerticalScrollOffset()
                        updateToolbarShadow()
                    }
                })
            }
        updateToolbarShadow()
        setDividerHeight(0)
    }

    private fun updateToolbarShadow() {
        val isVisible = listScrollY > 0
        if (lastIsVisible != isVisible) {
            (activity as? AppCompatActivity)?.apply {
                supportActionBar?.apply {
                    elevation = if (isVisible) {
                        getDimenPx(R.dimen.dp2).toFloat()
                    } else {
                        0f
                    }
                }
            }
            lastIsVisible = isVisible
        }
    }
}
